web_poet/requests.py

```diff
@@ -0,0 +1,184 @@
+        factory=HttpRequestHeaders, converter=HttpRequestHeaders
+    )
+    body: HttpRequestBody = attrs.field(
+        factory=HttpRequestBody, converter=HttpRequestBody
+    )
+
+
+async def _perform_request(request: HttpRequest) -> HttpResponse:
+    """Given a :class:`~.Request`, execute it using the **request implementation**
+    that was set in the ``web_poet.request_backend_var`` :mod:`contextvars`
+    instance.
+
+    .. warning::
+        By convention, this function should return a :class:`~.HttpResponse`.
+        However, the underlying downloader assigned in
+        ``web_poet.request_backend_var`` might change that, depending on
+        how the framework using **web-poet** implements it.
+    """
+
+    logger.info(f"Requesting page: {request}")
+
+    try:
+        request_backend = request_backend_var.get()
+    except LookupError:
+        raise RequestBackendError(
+            "Additional requests are used inside the Page Object but the "
+            "current framework has not set any HttpRequest Backend via "
+            "'web_poet.request_backend_var'"
+        )
+
+    response_data: HttpResponse = await request_backend(request)
+    return response_data
+
+
+class HttpClient:
+    """A convenient client to easily execute requests.
+
+    By default, it uses the request implementation assigned in the
+    ``web_poet.request_backend_var`` which is a :mod:`contextvars` instance to
+    download the actual requests. However, it can easily be overridable by
+    providing an optional ``request_downloader`` callable.
+
+    Providing the request implementation by dependency injection would be a good
+    alternative solution when you want to avoid setting up :mod:`contextvars`
+    like ``web_poet.request_backend_var``.
+
+    In any case, this doesn't contain any implementation about how to execute
+    any requests fed into it. When setting that up, make sure that the downloader
+    implementation returns a :class:`~.HttpResponse` instance.
+    """
+
+    def __init__(self, request_downloader: Callable = None):
+        self.request_downloader = request_downloader or _perform_request
+
+    async def request(
+        self,
+        url: str,
+        *,
+        method: str = "GET",
+        headers: Optional[Headers] = None,
+        body: Optional[Body] = None,
+    ) -> HttpResponse:
+        """This is a shortcut for creating a :class:`HttpRequest` instance and executing
+        that request.
+
+        A :class:`~.HttpResponse` instance should then be returned.
+
+        .. warning::
+            By convention, the request implementation supplied optionally to
+            :class:`~.HttpClient` should return a :class:`~.HttpResponse` instance.
+            However, the underlying implementation supplied might change that,
+            depending on how the framework using **web-poet** implements it.
+        """
+        req = HttpRequest(url, method, headers or {}, body or b"")
```

Comment:
let's use keyword arguments here

Reply:
Addressed this in https://github.com/scrapinghub/web-poet/pull/22/commits/6a53b90eb8cfa34d39bc7556d87f103480fd40c4.

Moreover, to clarify, did you mean to also enforce keyword args in the classes via `@attr.define(..., kw_only=True)`?

Reply:
I was thinking only about changing this example. As for making all attributes kw-only, url would be nice to keep positional.

Reply:
Got it. Nonetheless, I think it's worth changing the other non-URL init params to be kw-only to enforce correctness in case the args have unintentionally swapped positions. I've updated this in https://github.com/scrapinghub/web-poet/pull/22/commits/d88a0e5ceeb195d9adb8dbafb89ee338ad06b545